src/pal/inc/pal.h

```diff
@@ -5663,6 +5663,11 @@ PAL_FreeExceptionRecords(
   IN EXCEPTION_RECORD *exceptionRecord, 
   IN CONTEXT *contextRecord);
 
+VOID
+AllocateExceptionRecords(
```

Comment:
I don't like exposing the AllocateExceptionRecords from PAL. Nothing out of PAL needs this functionality. That's why I've suggested adding the Set method to the exception so that you can allocate the records in the PAL code and just set them on the exception (or without the Set, just update the members).

Reply:
The point is that PAL should expose the minimal possible surface and the record allocation is internal implementation detail of the PAL.